docs/UPGRADING.md

```diff
@@ -1,5 +1,25 @@
 # Upgrading
 
+## v3 to v4
+
+Kiam changed significantly between v3.X and v4.0. Breaking changes are:
+
+- The role policy is now applied after the role ARN has been resolved, this may cause compatibility issues with existing `iam.amazonaws.com/permitted` restrictions.
+- StatsD metrics have been removed.
+- The agent gRPC keepalve flags have been renamed.
+
+We would suggest upgrading in the following way:
+
+1. TODO
+2. If you still require StatsD metrics you may need to look at something like [veneur-prometheus](https://github.com/stripe/veneur/tree/master/cmd/veneur-prometheus) to scrape the /metrics endpoint and push them to StatsD.
```

Comment:
@pingles I am not sure what to say here, this the only possible prom->statsd route that I have hear of, any other ideas?

Reply:
I'm not aware of any others. @Joseph-Irving aware of anything worth calling out?

If not, I'm hopeful that it affects a tiny number of users. Thanks for putting in something.